// angular-file-picker/bundles/file-picker.umd.js
(function (global, factory) {
	typeof exports === 'object' && typeof module !== 'undefined' ? factory(exports, require('@angular/core')) :
	typeof define === 'function' && define.amd ? define(['exports', '@angular/core'], factory) :
	(factory((global.ng = global.ng || {}, global.ng.filePicker = global.ng.filePicker || {}),global.ng.core));
}(this, (function (exports,_angular_core) { 'use strict';

var PickedFileImpl = (function () {
    function PickedFileImpl(_lastModifiedDate, _name, _size, _type, _dataURL) {
        this._lastModifiedDate = _lastModifiedDate;
        this._name = _name;
        this._size = _size;
        this._type = _type;
        this._dataURL = _dataURL;
    }
    Object.defineProperty(PickedFileImpl.prototype, "lastModifiedDate", {
        get: function () {
            return this._lastModifiedDate;
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(PickedFileImpl.prototype, "name", {
        get: function () {
            return this._name;
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(PickedFileImpl.prototype, "size", {
        get: function () {
            return this._size;
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(PickedFileImpl.prototype, "type", {
        get: function () {
            return this._type;
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(PickedFileImpl.prototype, "dataURL", {
        get: function () {
            return this._dataURL;
        },
        enumerable: true,
        configurable: true
    });
    return PickedFileImpl;
}());

var FilePickerDirective = (function () {
    function FilePickerDirective(el, renderer) {
        this.el = el;
        this.renderer = renderer;
        this.accept = '';
        this.filePick = new _angular_core.EventEmitter();
    }
    FilePickerDirective.prototype.ngOnInit = function () {
        var _this = this;
        this.input = this.renderer.createElement(this.el.nativeElement.parentNode, 'input');
        this.renderer.setElementAttribute(this.input, 'type', 'file');
        this.renderer.setElementAttribute(this.input, 'accept', this.accept);
        this.renderer.setElementStyle(this.input, 'display', 'none');
        this.renderer.listen(this.input, 'change', function (event) {
            if (event.target.files.length < 1) {
                return;
            }
            _this.readFile(event.target.files[0]);
        });
    };
    FilePickerDirective.prototype.browse = function () {
        this.renderer.invokeElementMethod(this.input, 'click');
    };
    FilePickerDirective.prototype.readFile = function (file) {
        var _this = this;
        var reader = new FileReader();
        reader.onload = function (loaded) {
            var fileReader = loaded.target;
            var pickedFile = new PickedFileImpl(file.lastModifiedDate, file.name, file.size, file.type, fileReader.result);
            _this.filePick.emit(pickedFile);
        };
        reader.readAsDataURL(file);
    };
    return FilePickerDirective;
}());
FilePickerDirective.decorators = [
    { type: _angular_core.Directive, args: [{
                selector: '[ngFilePicker]'
            },] },
];
/** @nocollapse */
FilePickerDirective.ctorParameters = function () { return [
    { type: _angular_core.ElementRef, },
    { type: _angular_core.Renderer, },
]; };
FilePickerDirective.propDecorators = {
    'accept': [{ type: _angular_core.Input },],
    'filePick': [{ type: _angular_core.Output },],
    'browse': [{ type: _angular_core.HostListener, args: ['click',] },],
};

var FilePickerModule = (function () {
    function FilePickerModule() {
    }
    return FilePickerModule;
}());
FilePickerModule.decorators = [
    { type: _angular_core.NgModule, args: [{
                declarations: [FilePickerDirective],
                exports: [FilePickerDirective]
            },] },
];
/** @nocollapse */
FilePickerModule.ctorParameters = function () { return []; };

exports.FilePickerModule = FilePickerModule;
exports.FilePickerDirective = FilePickerDirective;

Object.defineProperty(exports, '__esModule', { value: true });

})));
