// angular-file-picker/src/file-picker.directive.js
import { Directive, ElementRef, EventEmitter, Input, HostListener, Output, Renderer } from '@angular/core';
import { PickedFileImpl } from './picked-file-impl';
var FilePickerDirective = (function () {
    function FilePickerDirective(el, renderer) {
        this.el = el;
        this.renderer = renderer;
        this.accept = '';
        this.filePick = new EventEmitter();
    }
    FilePickerDirective.prototype.ngOnInit = function () {
        var _this = this;
        this.input = this.renderer.createElement(this.el.nativeElement.parentNode, 'input');
        this.renderer.setElementAttribute(this.input, 'type', 'file');
        this.renderer.setElementAttribute(this.input, 'accept', this.accept);
        this.renderer.setElementStyle(this.input, 'display', 'none');
        this.renderer.listen(this.input, 'change', function (event) {
            if (event.target.files.length < 1) {
                return;
            }
            _this.readFile(event.target.files[0]);
        });
    };
    FilePickerDirective.prototype.browse = function () {
        this.renderer.invokeElementMethod(this.input, 'click');
    };
    FilePickerDirective.prototype.readFile = function (file) {
        var _this = this;
        var reader = new FileReader();
        reader.onload = function (loaded) {
            var fileReader = loaded.target;
            var pickedFile = new PickedFileImpl(file.lastModifiedDate, file.name, file.size, file.type, fileReader.result);
            _this.filePick.emit(pickedFile);
        };
        reader.readAsDataURL(file);
    };
    return FilePickerDirective;
}());
export { FilePickerDirective };
FilePickerDirective.decorators = [
    { type: Directive, args: [{
                selector: '[ngFilePicker]'
            },] },
];
/** @nocollapse */
FilePickerDirective.ctorParameters = function () { return [
    { type: ElementRef, },
    { type: Renderer, },
]; };
FilePickerDirective.propDecorators = {
    'accept': [{ type: Input },],
    'filePick': [{ type: Output },],
    'browse': [{ type: HostListener, args: ['click',] },],
};
